### DISCUSSION TITLE
Disable the `unicorn/prefer-array-find` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#230] [General]
There is a proposal to disable the `unicorn/prefer-array-find` rule.

> `unicorn/prefer-array-find`: this forbids this elegant way of filtering out nullish values: `Array.filter(Boolean)`
> \- @edgardz

**Comment by leroykorterink** [comment]
We should not disable this rule because [Array#find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) and [Array#findLast()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast) breaks the loop as soon as it finds a match and doesn't create a new array.

> this forbids this elegant way of filtering out nullish values: Array.filter(Boolean)

I've not seen this issue before, can you let me know how I this can be reproduced? Please see [the documentation](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-find.md) of the rule.

**Reply by ThaNarie** [reply]
> this forbids this elegant way of filtering out nullish values: Array.filter(Boolean)

When testing this, indeed it's still allowed:

```ts
const [result] = [undefined, 1, false].filter(Boolean);
const items = [undefined, 1, false].filter(Boolean);
```

eslint fixes the top one, but leaves the bottom one alone and is happy about that
```ts
const result = [undefined, 1, false].find(Boolean); // changed to using `find`, since only 1 result was used
const items = [undefined, 1, false].filter(Boolean);
```

@edgardz can we close this as incorrect? Or are we missing something? :)

**Reply by edgardz** [reply]
Very minor. Close it.
